### DISCUSSION TITLE
How to allow nullable/empty session variable

### DISCUSSION THREAD

**Discussion by DenisVASI9** [hasura/graphql-engine#8554] [Question] [answered]
Hi all, I have a header, x-hasura-var-id. Sometimes I don't want to pass it and expect it to be treated as null or an empty string, but it gives me an error that the header is not found. Right now I'm just passing an empty string as a stub, is there a better way?

**Comment by BenoitRanque** [comment, marked as answer]
Currently, if your query permissions require an environment variable not provided by your session you will face an error.

If you have different users types, where some do and some don't require the variables, you should probably define them as different roles.
